presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/TransformExistsApplyToCorrelatedJoin.java

```diff
@@ -108,6 +108,19 @@ public Result apply(ApplyNode parent, Captures captures, Context context)
             return Result.empty();
         }
 
+        /*
+        Empty correlation list indicates that the subquery contains no correlation symbols from the
+        immediate outer scope. The subquery might be either not correlated at all, or correlated with
+        symbols from further outer scope.
+        Currently, the two cases are indistinguishable.
+        To support the latter case, the ApplyNode with empty correlation list is rewritten to default
+        aggregation, which is inefficient in the rare case of uncorrelated EXISTS subquery,
+        but allows to successfully decorrelate a correlated EXISTS subquery.
```

Comment:
I would change `but allows to successfully decorrelate a correlated EXISTS subquery.` to:
```
but currently allows to successfully decorrelate a correlated EXISTS subquery.
```
Maybe add todo along the lines:
```
TODO: remove this condition when exploratory optimizer is implemented or
support for decorrelating joins is implemented in PlanNodeDecorrelator
```